File: src/Core/UWP/Sakuno.ING.Views.UWP.Common/Controls/EquipmentIcon.vb
Namespace Controls
    Public Class EquipmentIcon
        Inherits UserControl

        Private ReadOnly _image As New Image
        Public Sub New()
            Content = _image
        End Sub

        Private _id As Integer
        Public Property Id As Integer
            Get
                Return _id
            End Get
            Set(value As Integer)
                If _id <> value Then
                    _id = value
                    If DesignMode.DesignModeEnabled Then
                        _image.Source = New BitmapImage(New Uri($"ms-appx:///Sakuno.ING.Views.UWP.Common/Assets/Images/Equipment/{value}.png"))
                    Else
                        Dim stringId = "EquipmentIcon_Bitmap_" & value
                        Dim source As Object = Nothing
                        If Application.Current.Resources.TryGetValue(stringId, source) Then
                            _image.Source = DirectCast(source, BitmapImage)
                        Else
                            Dim s = New BitmapImage(New Uri($"ms-appx:///Sakuno.ING.Views.UWP.Common/Assets/Images/Equipment/{value}.png"))
                            Application.Current.Resources.Add(stringId, s)
                            _image.Source = s
                        End If
                    End If
                End If
            End Set
        End Property
    End Class
End Namespace

File: src/Core/UWP/Sakuno.ING.Views.UWP.Common/Controls/EquipmentIcon.cs
using System;
using Windows.ApplicationModel;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Media.Imaging;

namespace Sakuno.ING.Views.UWP.Controls
{
    public sealed class EquipmentIcon : UserControl
    {
        private readonly Image _image = new Image();
        public EquipmentIcon() => Content = _image;

        private int _id;
        public int Id
        {
            get => _id;
            set
            {
                if (_id != value)
                {
                    _id = value;
                    if (value == 0)
                        _image.Source = null;
                    else if (DesignMode.DesignModeEnabled)
                        _image.Source = new BitmapImage(new Uri($"ms-appx:///Sakuno.ING.Views.UWP.Common/Assets/Images/Equipment/{value}.png"));
                    else
                    {
                        string stringId = "EquipmentIcon_Bitmap_" + value;
                        if (Application.Current.Resources.TryGetValue(stringId, out object source))
                            _image.Source = source as BitmapImage;
                        else
                        {
                            var s = new BitmapImage(new Uri($"ms-appx:///Sakuno.ING.Views.UWP.Common/Assets/Images/Equipment/{value}.png"));
                            Application.Current.Resources[stringId] = s;
                            _image.Source = s;
                        }
                    }
                }
            }
        }
    }
}
